contracts/ServiceManagerToken.sol

```diff
@@ -149,21 +154,24 @@ contract ServiceManagerToken is GenericManager {
             for (uint256 i = 0; i < numAgents; ++i) {
                 // Copy actual bond values for each agent Id that has at least one slot in the updated service
                 if (agentParams[i].slots > 0) {
+                    // Check for the zero bond value
+                    if (agentParams[i].bond == 0) {
+                        revert ZeroValue();
+                    }
                     bonds[i] = agentParams[i].bond;
+                    // Wrap bonds with the BOND_WRAPPER value for the original ServiceRegistry contract
+                    agentParams[i].bond = BOND_WRAPPER;
                 }
-                // Wrap bonds with the BOND_WRAPPER value for the original ServiceRegistry contract
-                agentParams[i].bond = BOND_WRAPPER;
```

Comment:
Fixed a bug and check for the bond value to be bigger than zero when slots are bigger than zero as well